### PR TITLE
[MB-2978] Flash Message reducer & component

### PR DESCRIPTION
**Pull request by suzubara**
## Description

This PR adds a new `flash` reducer that can be used for setting (& clearing) flash messages that will be rendered on the Home page. I also added a `FlashMessage` component and corresponding HOC, which handles the following:
- `connectFlashMessage` connects to the `flash` reducer and renders `FlashMessage` only if there is a flash message defined (using the `key` property)
- `FlashMessage` renders the message in an `Alert` component, and also calls the `clearFlashMessage` action on unmount - this should guarantee flash messages are only shown to a user once, and will be cleared when the user navigates to another page.

I also replaced existing "flash" messages with this new component & reducer:
- Error if service member creation fails
- Error if unable to load user data (but user is logged in)
- Success when submitting a move
  - There was some existing logic to show a `PpmAlert` component here, but I discussed with the design team and we agreed to change to a more generic success message (which is [line 101 in the content doc](https://docs.google.com/spreadsheets/d/14_qlyhSN6-VjXmBLRtBHj2OUwy76NONOEUbYiRYQ2-w/edit#gid=1041135497))

## Reviewer Notes

Things I didn't do:
- Ideally, the `ConnectedFlashMessage` component is rendered at a more global level, such as MyMove/index.jsx (so it renders on every page, not just home). However right now there's some weird organization with the SignIn component for example, and I think we should clean up the global layout markup for the customer app generally, so I wanted to leave that for later.

To test:
- You can easily test the success flash message by submitting a move (with any shipment types) and verifying you see the success message on the home screen. It should then disappear once you click to another page.
- For the error messages, I'm not sure how to cause one of these errors in a real-life scenario. However, you can force the service member creation error if you log in as an existing service member, and then change [line 65 in `src/sagas/onboarding.js`](https://github.com/transcom/mymove/pull/5393/files#diff-6ceafd6b7d1973f6b2c2a2765bc9bdaf5dc05db40b6049076de7aaf04670fdd2L65) to `if (true)` ...then a server error will occur because it's attempting to create a service member for a user that already has one.

## Setup

```
make server_run
make client_run
```

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-2978) for this change

## Screenshots

image

image

### CODE REVIEW

File: src/testUtils.jsx [hunk 1]
```diff
@@ -23,9 +27,12 @@ MockProviders.propTypes = {
   // eslint-disable-next-line react/forbid-prop-types
   initialState: PropTypes.object,
   initialEntries: PropTypes.arrayOf(PropTypes.string),
+  // eslint-disable-next-line react/forbid-prop-types
```
Comment (sojeri) [inline src/testUtils.jsx:30]: APPROVED ;)

Kim's change will probably make it in in time that we don't even need this comment, but I wanted to add it anyway for others following the notifications.

**Comment by suzubara** [comment]
just to note there will be some conflicts with https://github.com/transcom/mymove/pull/5349 so I want to wait until that PR has been merged and then resolve them on this branch

**Review by lynzt — APPROVED** [review]
A few random test questions but LGTM...
I tested locally and I see the alert msg.

File: src/containers/FlashMessage/FlashMessage.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  flash: FlashMessageShape.isRequired,
+  clearFlashMessage: PropTypes.func.isRequired,
+};
+
+const connectFlashMessage = (Component) => {
+  const ConnectedFlashMessage = (props) => {
+    const { flash } = props;
+
+    // Only render flash if a key is defined
+    const showFlash = flash?.key;
```
[the remark this answers is not in the document]
Comment (suzubara) [reply src/containers/FlashMessage/FlashMessage.jsx:36]: good call, I updated the test to check for that

File: src/containers/FlashMessage/FlashMessage.test.jsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          message: 'This is a successful message!',
+          key: 'TEST_SUCCESS_FLASH',
+        }}
+        clearFlashMessage={mockClearFlash}
+      />,
+    );
+
+    expect(mockClearFlash).toHaveBeenCalledTimes(0);
+    wrapper.unmount();
+    expect(mockClearFlash).toHaveBeenCalledTimes(1);
```
[the remark this answers is not in the document]
Comment (suzubara) [reply src/containers/FlashMessage/FlashMessage.test.jsx:44]: so, this unit test is just making sure the mock function used for the `clearFlashMessage` prop gets called on `unmount`. it isn't changing any Redux state on its own. the actual `clearFlashMessage` action gets tested in the flash actions test file, and the reducer is also unit tested.

in order to test the full integrated functionality, we'd need to set up tests that render multiple pages & expose the `history` so we can make sure flash message state gets cleared when navigating to a new page. I definitely think this is worth adding tests for eventually, but probably not necessary at this point.

**Review by mariaclaratgolfo — APPROVED** [review]

**Pull request merged**